src/store/index.js

```diff
@@ -1,15 +1,56 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
+import { auth } from '../main';
 
 Vue.use(Vuex);
 
 export default new Vuex.Store({
   state: {
+    user: null,
+  },
+  getters: {
+    user: (state) => state.user,
   },
   mutations: {
+    setUser(state, payload) {
+      // this one is thorwing an error
+      //  state.user =[...payload]
+
+      // the following one will have issue if obj includes objs?
```

Comment:
comments should tell **why**, not **what**

Reply:
There really is no such thing as a deep copy in JS. If you really need a deep copy, writing one is fairly trivial.